test/test-integration.js

```diff
@@ -0,0 +1,79 @@
+/* global marionette, setup, test */
+
+'use strict';
+
+var assert = require('chai').assert;
+marionette.plugin('helper', require('marionette-helper'));
+
+marionette('gaia-header', function() {
+  var client = marionette.client({
+    profile: {
+      prefs: {
+        // Disable first time run UI
+        'browser.feeds.showFirstRunUI': false,
+        // Disable default browser check
+        'browser.shell.checkDefaultBrowser': false,
+        // Disable UI tutorial
+        'browser.uitour.enabled': false,
+        // Enable chrome debugging
+        'devtools.chrome.enabled': true,
+        'devtools.debugger.remote-enabled': true,
+
+        // Load integration test page on startup
+        'startup.homepage_welcome_url': __dirname + '/test-integration.html',
+
+        // Allow loading test resources oudside of test/ directory
+        // (e.g. bower-components)
+        'security.fileuri.strict_origin_policy': false,
+
+        // Enable web components
+        'dom.webcomponents.enabled': true,
+        // Enable touch events
+        'dom.w3c_touch_events.enabled': 1
+      }
+    },
+    desiredCapabilities: { raisesAccessibilityExceptions: true }
+  });
+
+  var headers = [{
+    selector: '#header-0'
+  }, {
+    selector: '#header-1'
+  }, {
+    selector: '#header-2'
+  }, {
+    selector: '#header-3'
+  }, {
+    selector: '#header-4'
+  }];
+
+  setup(function() {
+    headers.forEach(function(header) {
+      header.element = client.findElement(header.selector);
+    });
+  });
+
+  test('gaia-header is present', function() {
+    headers.forEach(function(header) {
+      assert.ok(header.element, header.selector);
+      assert.isTrue(header.element.displayed());
+    });
+  });
+
+  test('gaia-header after user action', function() {
+    ['click', 'tap'].forEach(function(action) {
+      headers.forEach(function(header) {
+        client.switchToShadowRoot(header.element);
+        var actionButton = client.helper.waitForElement('.action-button');
+        actionButton[action]();
+        client.switchToShadowRoot();
+      });
+
+      // Test additional (non-action header buttons)
+      var settingButton = client.helper.waitForElement('#setting');
+      settingButton[action]();
+      var customAnchor = client.helper.waitForElement('#custom');
+      customAnchor[action]();
+    });
```

Comment:
I can see in this test you are clicking and tapping some buttons, but I don't see any assertions being made. What is this test testing?